pandas/core/groupby/groupby.py

```diff
@@ -2453,6 +2454,9 @@ def pre_processor(vals: ArrayLike) -> tuple[np.ndarray, np.dtype | None]:
             elif is_timedelta64_dtype(vals.dtype):
                 inference = np.dtype("timedelta64[ns]")
                 out = np.asarray(vals).astype(float)
+            elif isinstance(vals, ExtensionArray) and is_float_dtype(vals):
+                inference = np.dtype(np.float64)
+                out = vals.to_numpy(dtype=float, na_value=np.nan)
```

Comment:
any reason for "float" instead of np.float64?

Reply:
actually I have picked this from the code around. Entire definition of `pre_processor` is like this.
https://github.com/pandas-dev/pandas/blob/a6943ae49f79452da6ca5f1c3e008ded4e0e2c85/pandas/core/groupby/groupby.py#L2435-L2459

Reply:
OK.  not a big deal, can probably do a followup to change a bunch of non-explicit dtypes to np.float64